bentoml/_internal/bento/bento.py

```diff
@@ -234,41 +236,26 @@ def save(
         return self
 
 
-class BentoStore(Store[OSBento]):
+class BentoStore(Store[SysPathBento]):
     def __init__(self, base_path: PathType):
-        super().__init__(base_path, OSBento)
+        super().__init__(base_path, SysPathBento)
 
 
-@attr.define(repr=False)
+@attr.define(repr=False, frozen=True)
 class BentoInfo:
-    service: str
     tag: Tag
-    bentoml_version: str
-    creation_time: datetime
-    labels: t.Dict[str, t.Any]
-    apis: t.Dict[str, "InferenceAPI"]
-    models: t.List[str]
+    service: str = attr.field(converter=lambda svc: svc._import_str)  # type: ignore[reportPrivateUsage]
+    labels: t.Dict[str, t.Any]  # TODO: validate user-provide labels
+    models: t.List[str]  # TODO: populate with model & framework info
+    bentoml_version: str = BENTOML_VERSION
+    creation_time: datetime = attr.field(factory=lambda: datetime.now(timezone.utc))
+
+    def __attrs_post_init__(self):
+        self.validate()
 
     def dump(self, stream: t.IO[t.Any]):
         return yaml.dump(self, stream, sort_keys=False)
 
-    @classmethod
-    def from_build_args(
-        cls, tag: Tag, svc: "Service", labels: t.Dict[str, t.Any], models: t.List[str]
-    ):
-        bento_info = cls(
-            service=svc._import_str,  # type: ignore[reportPrivateUsage]
-            tag=tag,
-            bentoml_version=BENTOML_VERSION,
-            creation_time=datetime.now(timezone.utc),
-            labels=labels,  # TODO: validate user provided labels
-            apis=svc._apis,  # type: ignore[reportPrivateUsage]
-            models=models,  # TODO: populate with model & framework info
-        )
-
-        bento_info.validate()
-        return bento_info
-
     @classmethod
     def from_yaml_file(cls, stream: t.IO[t.Any]):
```

Comment:
maybe it is type `BinaryIO` instead?

```suggestion
    def from_yaml_file(cls, stream: t.BinaryIO):
```

Reply:
Why? Does the method not support `TextIO`?

Reply:
Does passing a file in directly considered a TextIO? Passing files would be considered as a BinaryIO no?

Reply:
If you open a file with an encoding it's `TextIO`, I believe.